### ISSUE TITLE
Bugsnag: o.targetTouches is undefined

### ISSUE THREAD

**Issue by clifford-lc** [nolimits4web/swiper#2956]
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* Swiper Version: 4.0.7
* Vue Awesome Swiper Version: 3.1.3
* Platform/Target and Browser Versions: Chrome Mobile Android 7.0, iOS Safari iOS 12.1
* Live Link or JSFiddle/Codepen or website with issue: [https://www.optusperks.com.au](https://www.optusperks.com.au)

### What you did
We using the Vue Awesome Swiper package in our Vue project.

### Expected Behavior
No `bugsnag` errors.

### Actual Behavior
We are getting consistent bug snags errors `o.targetTouches is undefined` which point to a line in the swiper library:
`node_modules/swiper/dist/js/swiper.js:2623`
`touches.currentX = e.type === 'touchstart' ? e.targetTouches[0].pageX : e.pageX;`

![screen shot 2019-01-25 at 2 02 04 pm](https://user-images.githubusercontent.com/38636852/51722817-d44f3280-20aa-11e9-9330-017d1ce1af8c.png)

The best outcome here is that it's a type error which can be fixed to help suppress future warnings caused by the package.

**Comment by pdbreen** [comment]
Seeing same issue with direct use of Swiper v3.4.2  -  `TypeError · e.targetTouches is undefined`:  

```
ams.noSwipingClass))f.allowClick=!0;else if(!f.params.swipeHandler||N(e,f.params.swipeHandler)){var n=f.touches.currentX="touchstart"===e.type?e.targetTouches[0].pageX:e.pageX,i=f.touches.currentY="to
```

**Comment by versedi** [comment]
Firefox 62.0 for Android is also affected by this.

**Comment by matteo-rigon** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/targetTouches#Browser_compatibility

Apparently targetTouches property is not supported on Safari.

**Comment by spacebeers** [comment]
Having a similar issue with ios 13+.  

`undefined is not an object (evaluating 'targetTouches[0].pageX')`

**Comment by nolimits4web** [comment]
Would be great to see JSFiddle with the issue using core swiper.js (not some 3rd party Vue or React wrapper)

**Comment by spacebeers** [comment]
> Would be great to see JSFiddle with the issue using core swiper.js (not some 3rd party Vue or React wrapper)

https://codepen.io/SpaceBeers/full/GRRgRME

I _think_ the issue is caused from starting swiping from outside swiper but I'm not 100% on this yet.

**Comment by tom-sherman** [comment]
@versedi I have fixed the issue for iOS in #3259 but I wasn't able to replicate the issue because I don't have access to Firefox 62 or an Android tablet.
